File: BACK/Traffic Police/model/VehicleDTO.go
package model

import (
	"encoding/json"
	"io"
)

type VehicleDTO struct {
	Registration string `json:"registration"`
	Mark         string `json:"mark"`
	Model        string `json:"model"`
	OwnerId      string `json:"ownerId"`
	IsStolen     bool   `json:"isStolen"`
}

type Vehicles []*VehicleDTO

func (v *VehicleDTO) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(v)
}

func (v *Vehicles) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(v)
}

func (v *Vehicles) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	return d.Decode(v)
}

func (v *VehicleDTO) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	return d.Decode(v)
}

File: BACK/mup-vehicles/src/main/java/com/example/mupvehicles/dto/VehicleDto.java
package com.example.mupvehicles.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VehicleDto {

    private String id;

    private String mark;

    private String model;

    private String registration;

    private int year;

    private String color;

    private boolean isStolen;

    private String ownerId;
}
